### ISSUE TITLE
[clang-tidy] `readability-braces-around-statements` false positive with variable template lambda

### ISSUE THREAD

**Issue by FabianWolff** [llvm/llvm-project#64545]
Consider:
```c++
template <bool B>
inline constexpr auto foo = [] {
    if constexpr (B) {}
};

void bar() {
    foo<false>();
}
```

Expected behavior: No warnings, because every statement already has braces around it

Actual behavior ([Godbolt](https://godbolt.org/z/4oazMvx37)):
```
<source>:3:21: warning: statement should be inside braces [readability-braces-around-statements]
    3 |     if constexpr (B) {}
      |                     ^ 
      |                      {
1 warning generated.
```

**Comment by HerrCai0907** [comment]
candidate: https://reviews.llvm.org/D158480
